core/src/ecs/systems/ui/InputSystem.ts

```diff
@@ -214,6 +215,13 @@ export const InputSystem = ClientSystemBuilder({
         // update mouse position, the camera might have moved
         if (renderer) mouse = renderer.camera.toWorldCoords(mouseEvent);
 
+        // clear buffer if the window is not focused
+        if (!document.hasFocus()) {
+          bufferedDown.clear();
+          bufferedUp.clear();
+          return;
+        }
```

Comment:
clearing the buffer when document loses focus prevents the issue of your character walking in the same direction when you tab away